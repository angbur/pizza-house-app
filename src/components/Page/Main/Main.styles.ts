import styled from 'styled-components';

export const Box = styled.div`
  display: flex;
  flex-direction: column;
  margin-left: 114px;
`;
