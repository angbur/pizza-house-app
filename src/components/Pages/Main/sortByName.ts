import { Pizza } from 'types/Pizza';

export const sortByName = (data: Pizza[]) => data.sort((a, b) => compareFn(a, b));

const compareFn = (a: Pizza, b: Pizza) => {
  const nameA = a.name.toUpperCase();
  const nameB = b.name.toUpperCase();
  if (nameA < nameB) {
    return -1;
  }
  if (nameA > nameB) {
    return 1;
  }
  return 0;
};
