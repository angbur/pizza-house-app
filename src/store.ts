import { configureStore } from '@reduxjs/toolkit';
import orderReducer from 'components/Pages/Order/orderSlice';
import { pizzaApi } from 'services/pizza';

export const store = configureStore({
  reducer: {
    order: orderReducer,
    [pizzaApi.reducerPath]: pizzaApi.reducer,
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(pizzaApi.middleware),
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
