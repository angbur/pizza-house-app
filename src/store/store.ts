import { configureStore, Store } from '@reduxjs/toolkit';
import orderReducer from 'components/Pages/Order/orderSlice';
import dialogReducer from 'components/Organisms/Dialog/dialogSlice';
import { pizzaApi } from 'services/pizza';
import orderMiddleware from './middleware/orderMiddleware';

export const store: Store = configureStore({
  reducer: {
    order: orderReducer,
    dialog: dialogReducer,
    [pizzaApi.reducerPath]: pizzaApi.reducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware().concat(pizzaApi.middleware, orderMiddleware),
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
