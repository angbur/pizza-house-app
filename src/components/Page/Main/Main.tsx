import { Box } from './Main.styles';

const Main = () => {
  return (
    <main>
      <Box className='box'></Box>
    </main>
  );
};

export default Main;
