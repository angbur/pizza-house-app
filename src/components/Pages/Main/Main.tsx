import Typography from 'components/Atoms/Typography/Typography';
import { Box } from './Main.styles';

const Main = () => {
  return (
    <main>
      <Box className='box'>
        <Typography variant='main-title' color={'primary'}>
          Hello
        </Typography>
      </Box>
    </main>
  );
};

export default Main;
