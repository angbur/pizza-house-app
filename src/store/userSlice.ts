import { createSlice } from '@reduxjs/toolkit';
import { RootState } from './store';
import { toast } from 'react-toastify';
import { userApi } from 'services/user';

type User = {
  loggedIn: boolean;
  status: string | undefined;
  error: string | undefined;
  userId: string | undefined;
  login: string | undefined;
  token: string | undefined;
};

const initialState: User = {
  loggedIn: false,
  status: 'idle',
  error: undefined,
  userId: undefined,
  login: undefined,
  token: undefined,
};

const userSlice = createSlice({
  name: 'user',
  initialState,
  reducers: {
    logout: (state) => {
      state.loggedIn = false;
      state.userId = undefined;
      state.login = undefined;
      state.token = undefined;
      toast.success('Logged out!');
    },
  },
  extraReducers: (builder) => {
    builder.addMatcher(userApi.endpoints.login.matchFulfilled, (state, { payload }) => {
      state.loggedIn = true;
      state.token = payload.token;
      state.userId = payload.userId;
      state.login = payload.login;
    });
  },
});

export const { logout } = userSlice.actions;

export const selectName = (state: RootState) => state.user.user;

export const isLoggedIn = (state: RootState) => state.user.loggedIn;

export const selectResponseStatus = (state: RootState) => state.user.status;

export const selectResponseError = (state: RootState) => state.user.error;

export const selectLoggedInUser = (state: RootState) => state.user.user;

export default userSlice.reducer;
