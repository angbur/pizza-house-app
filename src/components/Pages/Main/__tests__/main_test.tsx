import renderer from 'react-test-renderer';
import ThemeProvider from 'components/Theme/ThemeContext';
import { store } from 'store/store';
import { Provider } from 'react-redux';
import Main from '../Main';

const Wrapper = () => {
  return (
    <ThemeProvider>
      <Provider store={store}>
        <Main />
      </Provider>
    </ThemeProvider>
  );
};

describe('Main Page', () => {
  test('should renders main page correctly', () => {
    const tree = renderer.create(<Wrapper />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
