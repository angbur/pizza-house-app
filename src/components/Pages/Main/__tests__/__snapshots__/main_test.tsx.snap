// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Page should renders main page correctly 1`] = `
<div
  className="sc-fEXmlR bWqDgE box"
>
  <main>
    <div
      className="sc-fnGiBr MacNp"
    >
      <h5
        className="item-title"
        style={
          Object {
            "color": "#FCFBFB",
            "fontFamily": "Source Sans Pro",
            "fontSize": "1rem",
            "fontWeight": "600",
            "letterSpacing": "1px",
            "lineHeight": "1.4",
            "textTransform": "uppercase",
          }
        }
      >
        Sort by
      </h5>
      <button
        className="sc-bcXHqe sc-gKPRtg sc-iBYQkv feNVH hUzBlu YZIpJ button-text-light btn-lg button"
        onClick={[Function]}
        style={
          Object {
            "fontSize": "1.05rem",
            "padding": "0.65rem 1.25rem",
          }
        }
      >
        <h5
          className="item-title"
          style={
            Object {
              "color": "#EB853E",
              "fontFamily": "Source Sans Pro",
              "fontSize": "1rem",
              "fontWeight": "600",
              "letterSpacing": "1px",
              "lineHeight": "1.4",
            }
          }
        >
          name
        </h5>
      </button>
      <button
        className="sc-bcXHqe sc-gKPRtg sc-iBYQkv feNVH hUzBlu YZIpJ button-text-light btn-lg button"
        onClick={[Function]}
        style={
          Object {
            "fontSize": "1.05rem",
            "padding": "0.65rem 1.25rem",
          }
        }
      >
        <h5
          className="item-title"
          style={
            Object {
              "color": "#FCFBFB",
              "fontFamily": "Source Sans Pro",
              "fontSize": "1rem",
              "fontWeight": "600",
              "letterSpacing": "1px",
              "lineHeight": "1.4",
            }
          }
        >
          price 
          <img
            height={8}
            src={
              Object {
                "ReactComponent": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "render": [Function],
                },
                "__esModule": true,
                "default": "arrow-down.svg",
              }
            }
          />
        </h5>
      </button>
      <button
        className="sc-bcXHqe sc-gKPRtg sc-iBYQkv feNVH hUzBlu YZIpJ button-text-light btn-lg button"
        onClick={[Function]}
        style={
          Object {
            "fontSize": "1.05rem",
            "padding": "0.65rem 1.25rem",
          }
        }
      >
        <h5
          className="item-title"
          style={
            Object {
              "color": "#FCFBFB",
              "fontFamily": "Source Sans Pro",
              "fontSize": "1rem",
              "fontWeight": "600",
              "letterSpacing": "1px",
              "lineHeight": "1.4",
            }
          }
        >
          price 
          <img
            height={8}
            src={
              Object {
                "ReactComponent": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "render": [Function],
                },
                "__esModule": true,
                "default": "arrow-up.svg",
              }
            }
          />
        </h5>
      </button>
    </div>
    <div
      className="sc-bjfHbI kWuuQk"
    />
  </main>
</div>
`;
